app/src/main/java/com/eveningoutpost/dexdrip/GcmListenerSvc.java

```diff
@@ -552,7 +557,8 @@ public void run() {
                             UserError.Log.wtf(TAG, "Exception processing rsom timestamp");
                         }
                     }
-
+                } else if (action.equals("libreBlock")) {
+                    HandleLibreBlock(payload);
```

Comment:
Doesn't this call `NFCReaderX.HandleGoodReading` which then in turn calls `GcmActivity.pushLibreBlock` which causes the received data to be duplicated? Its very important that inbound message doesn't generate the same outbound message.

Reply:
Messages should be unique by their timestamp.
In the function HandleLibreBlock we do the following:
1) We check if we have already seen this message (based on it's timestamp). If we did, we ignore it.
2) We save it in our DB.

3) On the function SendLibrereading() we check for a message with this timestamp again. if it is stored in our DB, we will not send it.
So, a message received from the network should not be sent again.